lib/classes/PluginManager.js

```diff
@@ -130,16 +130,32 @@ class PluginManager {
   }
 
   loadServicePlugins(servicePlugs) {
-    const servicePlugins = Array.isArray(servicePlugs) ? servicePlugs : [];
-
     // eslint-disable-next-line no-underscore-dangle
     module.paths = Module._nodeModulePaths(process.cwd());
 
+    const pluginsObject = this.parsePluginsObject(servicePlugs);
     // we want to load plugins installed locally in the service
-    if (this.serverless && this.serverless.config && this.serverless.config.servicePath) {
-      module.paths.unshift(path.join(this.serverless.config.servicePath, '.serverless_plugins'));
+    if (pluginsObject.localPath) {
+      module.paths.unshift(pluginsObject.localPath);
     }
-    this.loadPlugins(servicePlugins);
+    this.loadPlugins(pluginsObject.modules);
+  }
+
+  parsePluginsObject(servicePlugs) {
+    let localPath = (this.serverless && this.serverless.config &&
+      this.serverless.config.servicePath) &&
+      path.join(this.serverless.config.servicePath, '.serverless_plugins');
+    let modules = [];
+
+    if (Array.isArray(servicePlugs)) {
+      modules = servicePlugs;
+    } else if (servicePlugs && Array.isArray(servicePlugs.modules)) {
+      modules = servicePlugs.modules;
+      localPath = servicePlugs.localPath &&
+        typeof servicePlugs.localPath === 'string' ? servicePlugs.localPath : localPath;
```

Comment:
`_.isString()`

Reply:
I have replaced native calls to lodash isObject, isString and isArray.